internal/exec/stages/files/units.go

```diff
@@ -73,12 +75,12 @@ func (s *stage) createUnits(config types.Config) error {
 				if _, ok := presets[key]; ok {
 					presets[key].instances = append(presets[key].instances, instance)
 				} else {
-					presets[key] = &Preset{unitName, *unit.Enabled, true, []string{instance}}
+					presets[key] = &Preset{unitName, *unit.Enabled, true, []string{instance}, util.SystemdPresetPath(unit)}
```

Comment:
We're putting system and user units in the same namespace, which isn't what systemd does.  So, if a parent config specifies a `foo.service` system unit and a child specifies a `foo.service` user unit, the second will override the first.  If one config specifies both a `foo.service` system and user unit, config validation will fail.  (And currently, the code will get confused by instantiated units of system and user services with the same name.)

That's probably not a huge problem, but it's confusing.  We should probably track system and user units separately.  For config validation, `Unit` can do this by implementing `Keyed`.  And here, we'd need multiple maps, or a map of maps.

Reply:
changed /ignition/config/v3_4_experimental/types/unit.go to get a more distinct unit.key() value and use it